stan/math/prim/prob/wiener4_lccdf.hpp

```diff
@@ -0,0 +1,372 @@
+#ifndef STAN_MATH_PRIM_PROB_WIENER4_LCCDF_HPP
+#define STAN_MATH_PRIM_PROB_WIENER4_LCCDF_HPP
+
+#include <stan/math/prim/prob/wiener4_lcdf.hpp>
+
+namespace stan {
+namespace math {
+namespace internal {
+
+/**
+ * Log of probability of reaching the upper bound in diffusion process
+ *
+ * @tparam T_a type of boundary
+ * @tparam T_w type of relative starting point
+ * @tparam T_v type of drift rate
+ *
+ * @param a The boundary separation
+ * @param w The relative starting point
+ * @param v The drift rate
+ * @return log probability to reach the upper bound
+ */
+template <typename T_a, typename T_w, typename T_v>
+inline auto log_wiener_prob_hit_upper(const T_a& a, const T_v& v,
+                                      const T_w& w) {
+  using ret_t = return_type_t<T_a, T_w, T_v>;
+  const auto neg_v = -v;
+  const auto one_m_w = 1.0 - w;
+  if (fabs(v) == 0.0) {
+    return ret_t(log1m(one_m_w));
```

Comment:
This should just be `log(w)` because `log1m(1 - w) = log(1 - (1 - w)) = log(w)`.